app/controllers/reports_controller.rb

```diff
@@ -75,10 +73,7 @@ def report_params
     params.require(:report).permit(:title, :memo)
   end
 
-  def check_owner
-    return if @report.user == current_user
-
-    flash[:alert] = t('controllers.common.user_only', name: Comment.model_name.human)
-    redirect_to @report
+  def confirm_the_owner
+    current_user.reports.find(params[:id])
```

Comment:
このコードをset_reportにしてしまった方がスマートです（SQLの発行も一回で済む）

```suggestion
  def set_report
    @report = current_user.reports.find(params[:id])
```

Reply:
はい、set_reportとなるよう修正しました!